src/main/kotlin/uk/gov/justice/digital/hmpps/breachnoticeapi/service/NDeliusIntegrationService.kt

```diff
@@ -0,0 +1,21 @@
+package uk.gov.justice.digital.hmpps.breachnoticeapi.service
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import org.springframework.web.reactive.function.client.WebClient
+
+@Service
+class NDeliusIntegrationService(
+  private val webClient: WebClient,
+  @Value("\${ndelius-integration-api.url}") val ndeliusIntegrationApiUrl: String,
+) {
+  fun getCrnForBreachNoticeUuid(breachNoticeId: String): NDeliusCrn? = webClient.get()
+    .uri(ndeliusIntegrationApiUrl + "/case/{breachNoticeId}", breachNoticeId)
+    .retrieve()
+    .bodyToMono(NDeliusCrn::class.java)
+    .block()
```

Comment:
The return type here is nullable, but it looks like it will always either return or throw.  Would you need to add an `onErrorContinue` to handle a 404 and return null?  Or should we just make the return type non-nullable?

```suggestion
  fun getCrnForBreachNoticeUuid(breachNoticeId: String): NDeliusCrn? = webClient.get()
    .uri(ndeliusIntegrationApiUrl + "/case/{breachNoticeId}", breachNoticeId)
    .retrieve()
    .bodyToMono(NDeliusCrn::class.java)
    .block()
```